_posts/2020-08-20-codestarts-preview.adoc

```diff
@@ -0,0 +1,114 @@
+---
+layout: post
+title: 'Codestarts - Preview the new way to start coding with Quarkus'
+date: 2020-08-20
+tags: extension codestart quickstart
+synopsis: The idea is simple, you pick your extensions and they are able to provide example code to get you started with it :)
+author: adamevin
+---
+
+= Preview the new way to start coding with Quarkus
+
+*Have you ever generated a Quarkus project using the Maven plugin or code.quarkus.io?*
+
+If you did, then you might have wondered why, with the breadth of Quarkus extensions,
+we would only let you generate a REST API example application. Of course,
+you can pick extensions which are going to be in the dependencies of your app,
+this is pretty cool because some extensions are providing new features without any extra code. However,
+some extensions also require configuration and/or code to work, making it pretty limited as it is.
+
+== Introducing our new code generation system
+
+This is why I am thrilled to introduce the preview of our new code generation system in Quarkus `1.7.0.Final`.
+It is called Codestart. The idea is simple, you pick your extensions and they are able to provide example code to get you started with it :)
+
+This preview version is just the beginning and includes three example codestarts:
+
+- The legendary RestEasy example in Java, Kotlin and Scala, when you pick the RestEasy extension
+- The lovely and sweet template engine Qute in Java, when you pick the Qute extension
+- The tough and strong Quarkus Command Mode in Java, when you don’t pick any extension that provides an example
+
+== Wanna give it a try?
+
+We added a feature flag in the Quarkus Maven plugin to create your application:
+[source,shell,subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:create -DcodestartsEnabled`
```

Comment:
You say it's called `Codestart` singular above but the option is called `codestarts` with an s?

Reply:
You are right, `-DCodestartEnabled` would have been better.

Reply:
Well, you call it Codestarts elsewhere in the document so that's not a big issue.

Reply:
I would say that, "Codestart" is the feature, then each item is a "codestart", that can be plural, the feature is not supposed to be used as plural, but now I guess it's too late for that..

Reply:
then let's say the feature is `Codestarts`, I make sure the article is consistent and rebase.

Reply:
Ok, rebased. So `Codestarts` is the feature with the First letter upper cap, codestarts means a group of codestart items, a codestart is just one (without cap).